src/components/Table/utils/TableState.ts

```diff
@@ -29,11 +29,16 @@ export type SelectedState = "checked" | "unchecked" | "partial";
  */
 export class TableState {
   // A set of just row ids, i.e. not row.kind+row.id
-  private readonly collapsedRows = new ObservableSet<string>();
+  private readonly collapsedRows = new ObservableSet<string>([]);
   private persistCollapse: string | undefined;
-  private readonly selectedRows = new ObservableMap<string, SelectedState>();
+  private readonly rowSelectedState = new ObservableMap<string, SelectedState>();
+  // Keep a copy of all the selected GridDataRows - Do not include custom row kinds such as `header`, `totals`, etc.
+  // This allows us to keep track of rows that were selected, but may no longer be defined in `GridTable.rows`
```

Comment:
Maybe work the term "kept" into here, like just the end "i.e. what we call 'kept' rows"